managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedCursorImpl.java

```diff
@@ -1347,7 +1347,7 @@ public Set<? extends Position> asyncReplayEntries(final Set<? extends Position>
     @Override
     public Set<? extends Position> asyncReplayEntries(Set<? extends Position> positions,
             ReadEntriesCallback callback, Object ctx, boolean sortEntries) {
-        List<Entry> entries = Lists.newArrayListWithExpectedSize(positions.size());
+        List<Entry> entries = Collections.synchronizedList(new ArrayList<>(positions.size()));
```

Comment:
Why should local variables care about concurrency?

Reply:
We will invoke `entries.add` method at callback in different thread.